corehq/apps/translations/integrations/transifex/views.py

```diff
@@ -493,6 +495,52 @@ def post(self, request, *args, **kwargs):
         return self.get(request, *args, **kwargs)
 
 
+@method_decorator([toggles.APP_TRANSLATIONS_WITH_TRANSIFEX.required_decorator()], name='dispatch')
+class MigrateTransifexProject(BaseTranslationsView):
+    page_title = ugettext_lazy('Migrate Project')
+    urlname = 'migrate_transifex_project'
+    template_name = 'migrate_project.html'
+
+    def section_url(self):
+        return reverse(MigrateTransifexProject.urlname, args=self.args, kwargs=self.kwargs)
+
+    @property
+    @memoized
+    def form(self):
+        if self.request.POST:
+            return MigrateTransifexProjectForm(self.domain, self.request.POST, self.request.FILES)
+        else:
+            return MigrateTransifexProjectForm(self.domain)
+
+    @property
+    def page_context(self):
+        context = super(MigrateTransifexProject, self).page_context
+        if context['transifex_details_available']:
+            context['migration_form'] = self.form
+        return context
+
+    def _perform_request(self):
+        migrator = self.form.migrator
+        # ToDo: Check if you can pass form.migrator instead of its attrs and then re-init it
```

Comment:
@esoergel @sravfeyn 
Anyone has suggestion on this?
my concern is that if i pass that object and its serialised, it has two apps on it that are cached properties([target app](https://github.com/dimagi/commcare-hq/pull/24652/files#diff-9839a5958aa3803d64dbca2fea8ed3b7R143) and [source app](https://github.com/dimagi/commcare-hq/pull/24652/files#diff-9839a5958aa3803d64dbca2fea8ed3b7R85)) and they would be serialised too, making huge args to be send over to celery. So i chose to just re-instantiate the migrator in the task.

Reply:
I like that logic.  It also makes `migrate_project_on_transifex` a bit easier to conceptualize on its own, since it accepts only relatively simple args, you can largely ignore how it's called.

Can `uploaded_resource_id_mappings` get big?

Reply:
Thanks @esoergel 
`uploaded_resource_id_mappings` is a tuple, (type, old id, new id). so the number of that would be the number forms + modules in an app. Can stretch for big apps. For ICDS it's 97 right now which i think is fine.